express-notes-api/index.js

```diff
@@ -0,0 +1,88 @@
+const express = require('express');
+const notesObject = require('./data.json');
+const fs = require('fs');
+const app = express();
+app.listen(3000);
+const jsonParser = express.json();
+app.use(jsonParser);
+
+app.get('/api/notes', (req, res) => {
+  const notesArray = [];
+  for (const note in notesObject.notes) {
+    notesArray.push(notesObject.notes[note]);
+  }
+  res.status(200).json(notesArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = req.params.id;
+  if (parseInt(id) < 0 || isNaN(id)) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (notesObject.notes[id]) {
+    res.status(200).json(notesObject.notes[id]);
+  } else if (!notesObject.notes[id]) {
```

Comment:
Since the second is the logical opposite of the first, might as well just make it an `else`.